protocols/kad/src/query/peers/closest.rs

```diff
@@ -192,7 +203,7 @@ impl ClosestPeersIter {
             };
             self.closest_peers.entry(distance).or_insert(peer);
 
-            progress = self.closest_peers.keys().next() == Some(&distance) || progress;
+            progress = distance < cur_range || progress;
```

Comment:
We also need to make sure that the peers with `distance < cur_range` are new (not already in `self.closest_peers`). If all `closer_peers` are already known, the query doesn't progress